site/src/components/Popover/Popover.tsx

```diff
@@ -41,32 +38,40 @@ const PopoverContext = createContext<PopoverContextValue | undefined>(
   undefined,
 );
 
-export interface PopoverProps {
-  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
+type BasePopoverProps = {
+  children: ReactNode;
   mode?: TriggerMode;
-  isDefaultOpen?: boolean;
-}
+};
 
-export const Popover: FC<PopoverProps> = ({
-  children,
-  mode,
-  isDefaultOpen,
-}) => {
+type UncontrolledPopoverProps = BasePopoverProps & {
+  open?: undefined;
+  onOpenChange?: undefined;
+};
+
+type ControlledPopoverProps = BasePopoverProps & {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+};
+
+export type PopoverProps = UncontrolledPopoverProps | ControlledPopoverProps;
+
+export const Popover: FC<PopoverProps> = (props) => {
   const hookId = useId();
-  const [isOpen, setIsOpen] = useState(isDefaultOpen ?? false);
-  const triggerRef = useRef<HTMLElement>(null);
+  const [open, setOpen] = useState(false);
```

Comment:
Just to avoid some confusion, could the variables be named to indicate that `open` and its state setter are specifically for when the component doesn't receive controlled state from outside?

Something like `internalIsOpen`?

Reply:
Used the following names to make it more descriptive as possible `uncontrolledOpen, setUncontrolledOpen`